### PR TITLE
Isaacs/fix tests and fail properly

### PR DESCRIPTION
**Pull request by isaacs**
Set the exit code properly when a command fails.

Fix the failure that our CI wasn't detecting for this reason.

### CODE REVIEW

File: test/lib/npm.js [hunk 1]
```diff
@@ -323,6 +323,9 @@ t.test('npm.load', t => {
       ])
       t.same(consoleLogs, [['@foo']])
     })
+
+    // need this here or node 10 will improperly end the promise ahead of time
+    await new Promise((res) => setTimeout(res))
```
Comment (ruyadorno) [inline test/lib/npm.js:328]: this is crazy hacky, would love to learn about the edge case going on here
